frontend/src/lib/components/apps/editor/settingsPanel/script/BackgroundScriptSettings.svelte

```diff
@@ -1,10 +1,25 @@
 <script lang="ts">
 	import ScriptSettingHeader from './shared/ScriptSettingHeader.svelte'
-	import type { HiddenInlineScript } from '$lib/components/apps/types'
+	import type { AppViewerContext, HiddenInlineScript } from '$lib/components/apps/types'
 	import ScriptRunConfiguration from './shared/ScriptRunConfiguration.svelte'
 	import BackgroundScriptTriggerBy from './shared/BackgroundScriptTriggerBy.svelte'
+	import { getContext } from 'svelte'
 
 	export let script: HiddenInlineScript
+	export let id: string
+
+	const { runnableComponents } = getContext<AppViewerContext>('AppViewerContext')
+
+	// If the script's autoRefresh value is different from the one in $runnableComponents, update $runnableComponents
+	$: if (
+		$runnableComponents?.[id]?.autoRefresh !== script.autoRefresh &&
+		script.autoRefresh !== undefined
+	) {
+		$runnableComponents[id] = {
+			...$runnableComponents[id],
+			autoRefresh: script.autoRefresh
+		}
+	}
```

Comment:
seems wrong to make it a reactive statement. Can that check not be done on change of script.autoRefresh ?